### PR TITLE
Video lisitng

### PR DESCRIPTION
**Pull request by DivyanshSareen**
added video listing page with video-video listing context
click on any category on homepage to get to the listing page
preview: 
image

### CODE REVIEW

File: src/context/category-context.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+const CategoryProvider = ({ children }) => {
+  const [categories, setCategories] = useState([]);
+
+  const getCategories = async () => {
+    let response = [];
+    try {
+      response = await axios.get("/api/categories");
+    } catch (error) {
+      console.log(error);
```
Comment (devAmlan) [inline src/context/category-context.js:13]: you can use toast to show warning/error to user instead of console.log

**Review by devAmlan — APPROVED** [review]
1. You can use toast instead of console.log in errors.
2. Code looks good and clean.
3. You can use a separate file for root CSS codes.

Good to merge.

File: src/styles/style.css [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+.card-content{
+    display: flex;
+    justify-content: baseline;
+    align-items: baseline;
+}
+
+.card-actions{
+    margin-left: auto;
+    margin-right: 0.5em;
```
Comment (bossmonkey98) [inline src/styles/style.css:77]: you can use pixel instead of em

**Review by bossmonkey98 — APPROVED** [review]

**Review by saurabhan — APPROVED** [review]
Once Again Beautiful UI and hover implementation, clean code, and loved how you break components. suggestion: define max-width so the UI can be more focused. great work

File: src/context/category-context.js (hunk 1 above)
Comment (DivyanshSareen) [reply src/context/category-context.js:13]: Will surely work on error handling and feedback soon

File: src/styles/style.css (hunk 2 above)
Comment (DivyanshSareen) [reply src/styles/style.css:77]: trying to keep it relative to fontsize. Will surely switch if UI breaks at some point.

**Comment by DivyanshSareen** [comment]
> Once Again Beautiful UI and hover implementation, clean code, and loved how you break components. suggestion: define max-width so the UI can be more focused. great work

Thank you for the review and suggestions will surely work on it

**Pull request merged**